packages/medusa/src/services/batch-job.ts

```diff
@@ -71,6 +136,243 @@ class BatchJobService extends TransactionBaseService<BatchJobService> {
+        }
+
+        if (batchJob.status !== BatchJobStatus.READY) {
+          throw new MedusaError(
+            MedusaError.Types.NOT_ALLOWED,
+            "Cannot confirm processing for a batch job that is not ready"
+          )
+        }
+
+        this.eventBus_
+          .withTransaction(transactionManager)
+          .emit(BatchJobService.Events.PROCESSING, {
+            id: batchJob.id,
+          })
+
+        return batchJob
+      }
+    )
+  }
+
+  async complete(batchJobOrId: string | BatchJob): Promise<BatchJob | never> {
+    return await this.atomicPhase_(async () => {
+      let batchJob: BatchJob = batchJobOrId as BatchJob
+      if (typeof batchJobOrId === "string") {
+        batchJob = await this.retrieve(batchJobOrId)
+      }
+
+      if (batchJob.status !== BatchJobStatus.PROCESSING) {
+        throw new MedusaError(
+          MedusaError.Types.INVALID_DATA,
+          `Cannot complete a batch job with status "${batchJob.status}". The batch job must be processing`
+        )
+      }
+
+      return await this.updateStatus(batchJob, BatchJobStatus.COMPLETED)
+    })
+  }
+
+  async cancel(batchJobOrId: string | BatchJob): Promise<BatchJob | never> {
+    return await this.atomicPhase_(async () => {
+      let batchJob: BatchJob = batchJobOrId as BatchJob
+      if (typeof batchJobOrId === "string") {
+        batchJob = await this.retrieve(batchJobOrId)
+      }
+
+      if (batchJob.status === BatchJobStatus.COMPLETED) {
+        throw new MedusaError(
+          MedusaError.Types.NOT_ALLOWED,
+          "Cannot cancel completed batch job"
+        )
+      }
+
+      return await this.updateStatus(batchJob, BatchJobStatus.CANCELED)
+    })
+  }
+
+  async ready(batchJobOrId: string | BatchJob): Promise<BatchJob | never> {
+    return await this.atomicPhase_(async () => {
+      let batchJob: BatchJob = batchJobOrId as BatchJob
+      if (typeof batchJobOrId === "string") {
+        batchJob = await this.retrieve(batchJobOrId)
+      }
+
+      if (batchJob.status !== BatchJobStatus.CREATED) {
+        throw new MedusaError(
+          MedusaError.Types.NOT_ALLOWED,
+          "Cannot mark a batch job as ready if the status is different that created"
+        )
+      }
+
+      return await this.updateStatus(batchJobOrId, BatchJobStatus.READY)
+    })
+  }
+
+  async processing(batchJobOrId: string | BatchJob): Promise<BatchJob | never> {
+    return await this.atomicPhase_(async () => {
+      let batchJob: BatchJob = batchJobOrId as BatchJob
+      if (typeof batchJobOrId === "string") {
+        batchJob = await this.retrieve(batchJobOrId)
+      }
+
+      if (batchJob.status !== BatchJobStatus.READY) {
+        throw new MedusaError(
+          MedusaError.Types.NOT_ALLOWED,
+          "Cannot mark a batch job as processing if the status is different that ready"
+        )
+      }
+
+      return await this.updateStatus(batchJob, BatchJobStatus.PROCESSING)
+    })
+  }
+
+  async pre_processing(
```

Comment:
todo: not sure how this sneaked in here but we use `camelCase` for function names :))

Reply:
Sure, i think i ve been changing too many times between both 😂 sorry